File: app/src/main/java/ru/geekbrains/notes/view/DialogueDelete.java
package ru.geekbrains.notes.view;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import ru.geekbrains.notes.MainActivity;
import ru.geekbrains.notes.data.CardSource;

public class DialogueDelete extends DialogFragment {


    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity())
                .setTitle("Are you sure you want to delete this note?")
                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        ((MainActivity) requireActivity()).onDialogResult(true);
                       // dismiss();
                    }
                })
                .setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dismiss();
                    }
                });
        setCancelable(false);
        return builder.create();
    }
}

File: app/src/main/java/ru/geekbrains/notes/view/DialogueDelete.kt
package ru.geekbrains.notes.view

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import ru.geekbrains.notes.model.CardData
import ru.geekbrains.notes.presenter.MainMainPresenter

class DialogueDelete(
    private val presenter: MainMainPresenter,
    private val cardData: CardData,
    private val position: Int
) : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireActivity())
            .setTitle("Are you sure you want to delete this note?")
            .setPositiveButton(
                "Yes"
            ) { dialogInterface, i ->
                presenter.deleteCard(position)
                presenter.notify(cardData)
                dismiss()
            }
            .setNegativeButton(
                "No"
            ) { dialogInterface, i -> dismiss() }
        isCancelable = false
        return builder.create()
    }
}
